02.calendar/cal.js

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env node
+import minimist from "minimist";
+
+const HEADER_WIDTH = 14;
+
+const run = () => {
+  const today = new Date();
+  const currentYear = today.getFullYear();
+  const currentMonth = today.getMonth() + 1;
+
+  const parseOptions = minimist(process.argv.slice(2), {
+    string: ["-y", "-m"],
+    alias: {
+      y: "year",
+      m: "month",
+    },
+    default: {
+      y: currentYear,
+      m: currentMonth,
+    },
+  });
+
+  const year = parseOptions.year;
+  const month = parseOptions.month;
+
+  return [
+    generateCalendarHeader(year, month),
+    "Su Mo Tu We Th Fr Sa",
+    ...generateMonthDates(year, month),
+  ].join("\n");
+};
+
+const generateCalendarHeader = (year, month) => {
+  const monthName = getFirstDateOfMonth(year, month).toLocaleString("en-US", {
+    month: "short",
+  });
+  return `${monthName} ${year}`.padStart(HEADER_WIDTH, " ");
+};
+
+const getFirstDateOfMonth = (year, month) => {
+  return new Date(year, month - 1, 1);
+};
+
+const generateMonthDates = (year, month) => {
+  const lastDay = getLastDayOfMonth(year, month);
+  const weeks = [];
+  let monthDates = "";
+  monthDates += generateBlankDays(year, month);
+
+  for (let day = 1; day <= lastDay; day++) {
+    monthDates += generateDayFormat(day);
+    if (isSaturday(year, month, day) || day === lastDay) {
+      monthDates = monthDates.trimEnd();
+      weeks.push(monthDates);
+      monthDates = "";
+    }
```

Comment:
```suggestion
    if (isSaturday(year, month, day) || day === lastDay) {
      weeks.push(monthDates);
      monthDates = "";
    } else {
      monthDates += " ";
    }
```

一度追加したものを削除すると二度手間なので、こんな感じで「必要であれば追加する（不要なら何もしない）」というロジックの方がスマートかな〜と思います。

Reply:
コメントありがとうございます！
ご提案のコードの方が書き方もスマートでそちらの方が俄然良いと思いました✨
こうしてみると`trimEnd`案はその場しのぎな感じが否めないです💦
特に大規模な処理では処理量が増えるにつれて負荷が徐々に蓄積していくと思うので、こうした小さな点も気を付けたいと思います🙏✨
（`trimEnd`のままではなんとなく気持ちが良くないので、こちらもコミットに含めてマージさせて頂こうと思います〜）